src/bench.c

```diff
@@ -214,24 +214,16 @@ char *benchmark_format(struct fmt_main *format, int salts,
 	return event_abort ? "" : NULL;
 }
 
-void benchmark_cps(int64 *crypts, clock_t time, char *buffer)
+void benchmark_cps(uint64_t crypts, clock_t time, char *buffer)
 {
-	unsigned int cps_hi, cps_lo;
-	int64 tmp;
-
-	tmp = *crypts;
-	mul64by32(&tmp, clk_tck);
-	cps_hi = div64by32lo(&tmp, time);
-
-	if (cps_hi >= 1000000)
-		sprintf(buffer, "%uK", cps_hi / 1000);
-	else
-	if (cps_hi >= 100)
-		sprintf(buffer, "%u", cps_hi);
-	else {
-		mul64by32(&tmp, 10);
-		cps_lo = div64by32lo(&tmp, time) % 10;
-		sprintf(buffer, "%u.%u", cps_hi, cps_lo);
+	unsigned int cps = crypts * clk_tck / time;
+	if (cps >= 1000000) {
```

Comment:
Is this a chance for overflow (the crypts*clk_tck ?  It was done as int64 in a prior life?  Should the cps be set to int64_t and the printf's changed to handle it?

Reply:
Solar opted to do it this way, and it avoids the printf nightmare with PRIu64 and such crap so I'm happy with it.

Reply:
Well, the math looks to be done uint64_t, with the final truncation to unsigned.  Yep, it does avoid the printf hell you get into when moving outside int / unsigned world.

Reply:
Maybe we have a very real problem in jumbo here for GPU/FPGA formats, where the c/s rate may reasonably exceed 4G. I merely didn't want to make unrelated changes along with the move from `int64` to `uint64_t`.

Reply:
We are only looking at the original core diff here, not the merged Jumbo diff. I believe Jumbo still has it right.

Reply:
$ git diff master bench.c
```diff
(...)
@@ -217,6 +591,13 @@ char *benchmark_format(struct fmt_main *format, int salts,
 void benchmark_cps(uint64_t crypts, clock_t time, char *buffer)
 {
        unsigned int cps = crypts * clk_tck / time;
+       uint64_t cpsl = crypts * clk_tck / time;
+
+       if (cpsl >= 1000000000000ULL) {
+               sprintf(buffer, "%uG", (uint32_t)(cpsl / 1000000000ULL));
+       } else if (cpsl >= 1000000000) {
+               sprintf(buffer, "%uM", (uint32_t)(cpsl / 1000000));
+       } else
        if (cps >= 1000000) {
                sprintf(buffer, "%uK", cps / 1000);
        } else if (cps >= 100) {
(...)
```
Jumbo adds support for M and G, with a non-conflicting diff as above.